app/components/layout/Navigation/MyBuilds/index.js

```diff
@@ -0,0 +1,194 @@
+        this.setState({ fetching: false });
+      }
+    });
+  }
+
+  componentWillUnmount() {
+    PusherStore.off("user_stats:change", this.handlePusherWebsocketEvent);
+  }
+
+  componentWillReceiveProps = (nextProps) => {
+    if (nextProps.viewer.scheduledBuilds || nextProps.viewer.runningBuilds) {
+      this.setCachedState({
+        scheduledBuildsCount: nextProps.viewer.scheduledBuilds.count,
+        runningBuildsCount: nextProps.viewer.runningBuilds.count
+      });
+    }
+  };
+
+  render() {
+    const buildsCount = this.state.runningBuildsCount + this.state.scheduledBuildsCount;
+    let badge;
+
+    if (buildsCount) {
+      badge = (
+        <Badge className={classNames("hover-lime-child", { "bg-lime": this.state.showingDropdown })}>
+          {buildsCount}
+        </Badge>
+      );
+    }
+
+    return (
+      <Dropdown width={320} className="flex" onToggle={this.handleBuildsDropdownToggle}>
+        <DropdownButton className={classNames("py0", { "lime": this.state.showingDropdown })}>
+          {'My Builds '}
+          <div className="xs-hide">
+            <ReactCSSTransitionGroup transitionName="transition-appear-pop" transitionEnterTimeout={200} transitionLeaveTimeout={200}>
+              {badge}
+            </ReactCSSTransitionGroup>
+          </div>
+          <Icon icon="down-triangle" style={{ width: 7, height: 7, marginLeft: '.5em' }} />
+        </DropdownButton>
+
+        {this.renderDropdown()}
+      </Dropdown>
+    );
+  }
+
+  renderDropdown() {
+    if (this.state.fetching) {
+      return this.renderSpinner();
+    } else if (this.props.viewer.user.builds.edges.length > 0) {
+      return this.renderBuilds();
+    } else {
+      return this.renderSetupInstructions();
+    }
+  }
+
+  renderBuilds() {
+    return (
+      <div>
+        <div className="px3 py2">
+          {this.props.viewer.user.builds.edges.map((edge) => <Build key={edge.node.id} build={edge.node} />)}
+        </div>
+        <div className="pb2 px3">
+          <Button href="/builds" theme="default" outline={true} className="center" style={{ width: "100%" }}>More Builds</Button>
+        </div>
+      </div>
+    );
+  }
+
+  renderSetupInstructions() {
+    return (
+      <div className="px3 py2">
+        <div className="mb2">To have your builds appear here, make sure that your commit email address (e.g. <code className="border border-gray dark-gray" style={{ padding: '.1em .3em' }}>git config user.email</code>) is added and verified in your list of personal email addresses.</div>
+        <Button href="/user/emails" theme="default" outline={true} className="center" style={{ width: "100%" }}>Update Email Addresses</Button>
+      </div>
+    );
+  }
+
+  renderSpinner() {
+    return (
+      <div className="px3 py2 center">
+        <Spinner />
+      </div>
+    );
+  }
+
+  handleBuildsDropdownToggle = (visible) => {
+    this.setState({ showingDropdown: visible });
+  };
+
+  handlePusherWebsocketEvent = (payload) => {
```

Comment:
This code is a little weird, but my next PR will probably change all these bits.